sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -188,6 +191,10 @@ def __init__(
             HttpLoggingPolicy(**kwargs),
         ]
 
+        if self.aad_credentials:
+            scopes = base.create_scope_from_url(self.url_connection)
```

Comment:
Are scopes constructed the same way in all clouds?

Reply:
For the most part yes, they take whatever prefix comes before the port and assign `/.default`. Only difference is for Cosmos we utilize the user account endpoint. Here is Storage SDK scopes for comparison (file is constants.py under _shared directory for blob sdk): 
<img width="352" alt="image" src="https://user-images.githubusercontent.com/30335873/159549168-8df9c726-f013-4704-a62c-fe8ebd8c8615.png">

Reply:
Are there scenario where customers use custom DNS names (either supported by the service - which I don't believe cosmos provides) or by custom proxies?

Reply:
We generally do not recommend using DNS since it can create issues regardless, and I do believe you'd need the database account endpoint as a scope (which means you'd need this information regardless), but I'll have to ask the Cosmos RBAC team to see how this works.

Reply:
Checked with the RBAC folks and it seems this isn't supported, so we should be good.

With that said, I do think the way I'm just replacing "443" with ".default" doesn't look so nice since we're searching the string for the 443... I thought of doing `return url.split(":")[0] + ":" + url.split(":")[1] + "/.default"` as well but not sure if that looks worse, any advice on this is appreciated!

Reply:
Just to go back Johan's original question - have we tested this with China and USGov clouds?
Does the AAD credentials work with Azure Stack or emulator - and if not - what kind of errors would I expect?

Reply:
@annatisch I asked Thomas Weiss from the security team and he has shared some of these answers with me; that data plane RBAC is available is sovereign clouds, but not in Azure Stack since CosmosDB isn't offered there.

For the emulator, I can forward you an email with the full explanation, but we do not support AAD since the emulator does not have an identity service, and is only meant to be used alongside a primary key (there's no way for you to set up an identity that can authorize against the emulator).

Reply:
Cool okay - so it looks like the only scenario we need to validate is the sovereign clouds :)

Reply:
I spoke to my team on AAD in the other SDKs and the use of sovereign clouds. Basically, the scope that we utilize for the token will always be this hardcoded convention using the user's database account, and does not depend on the cloud - it is a Cosmos specific scope.